### PR TITLE
Use interface for creating Ceph pools

### PR DESCRIPTION
**Pull request by joedborg**
This facilitates the ceph-client interface and removes a bunch of retrospectively redundant code.

Have tested:
```
kubectl get sc,po
NAME                                             PROVISIONER     AGE
storageclass.storage.k8s.io/ceph-ext4            csi-rbdplugin   72s
storageclass.storage.k8s.io/ceph-xfs (default)   csi-rbdplugin   72s

NAME                              READY   STATUS              RESTARTS   AGE
pod/csi-rbdplugin-attacher-0      1/1     Running             0          72s
pod/csi-rbdplugin-mxf6f           0/2     ContainerCreating   0          51s
pod/csi-rbdplugin-nb92n           1/2     ContainerCreating   0          51s
pod/csi-rbdplugin-provisioner-0   1/1     Running             0          72s
pod/csi-rbdplugin-z5fmh           2/2     Running             0          51s
```
We'll have to change the docs here: https://www.ubuntu.com/kubernetes/docs/storage to reflect the fact there are now 2 ceph interfaces:
```
juju add-relation kubernetes-master:ceph-storage ceph-mon:admin
juju add-relation kubernetes-master:ceph-client ceph-mon:client
```

### CODE REVIEW

**Comment by joedborg** [comment]
This would break `actions/create-rbd-pv` as it requires the installation of `ceph-common`

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 1]
```diff
@@ -1031,29 +1031,27 @@ def ceph_storage_privilege():
     reconfigure_apiserver(endpoint_from_flag('etcd.available'))
 
 
+@when('ceph-client.connected')
 @when('kubernetes-master.ceph.configured')
 @when_not('kubernetes-master.ceph.pool.created')
-def ceph_storage_pool():
+def ceph_storage_pool(ceph_client):
```
Comment (hyperbolic2346) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: passing parameters has been somewhat deprecated. Instead, use: `ceph_client = endpoint_from_flag('ceph-client.connected')

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 2]
```diff
@@ -1031,29 +1031,27 @@ def ceph_storage_privilege():
...
+    ]
 
-    ReplicatedPool(
-        name='ext4-pool',
-        service='admin',
-        replicas=3,
-        app_name=None,
-    ).create()
+    for pool in pools:
+        ceph_client.create_pool(
```
Comment (hyperbolic2346) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: Do we want to set a status message and wait on the flag to be set telling us the pools are created? What if the pool creation fails? Do we get notification of that in some way so we can surface an error to the user?

Comment (joedborg) [reply cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: Good point!  I'll add logging.

**Comment by joedborg** [comment]
@hyperbolic2346 Do you think the `create-rbd-pv` run action should install `ceph-common`?  Just so we're not installing it on all ceph clusters, by default?  Or do the majority of ceph users use this?

**Comment by hyperbolic2346** [comment]
It is the only reason we need it now and I would imagine the users of that action are relatively low. If we can do that cleanly, I'd be a fan.

**Comment by joedborg** [comment]
@hyperbolic2346 Just pushed that.

**Pull request merged**